store/sqlstore/channel_store.go

```diff
@@ -1840,12 +1840,15 @@ func (s SqlChannelStore) CountPostsSince(channelID string, since int64) (int64,
 // UpdateLastViewedAtPost sets a channel as unread for a user at the time of the post selected and update the MentionCount
 // it returns a channelunread so redux can update the apps easily.
 func (s SqlChannelStore) UpdateLastViewedAtPost(unreadPost *model.Post, userID string, mentionCount int) (*model.ChannelUnreadAt, *model.AppError) {
-
 	unread, appErr := s.CountPostsSince(unreadPost.ChannelId, unreadPost.CreateAt)
 	if appErr != nil {
 		return nil, appErr
 	}
 
+	if mentionCount == -1 {
```

Comment:
I can't find a good place to put a named sentinel value since this package can't access the `app` package and the `app` package can't reference the `sqlstore` package directly, but I'll add a few comments explaining that -1 is special here